src/adapters/RoomsJSONAdapter.js

```diff
@@ -74,9 +76,32 @@ export default class RoomsJSONAdapter extends RoomsAdapter {
     return Observable.create((observer) => {
       const data = !this.datasource[`${ID}-activities`] ? [] : this.datasource[`${ID}-activities`];
 
-      observer.next(data);
+      const formattedData = RoomsJSONAdapter.formatActivity(data);
+
+      observer.next(formattedData);
 
       observer.complete();
     });
   }
 }
+
+/**
+ * Activities could be objects with metadata (like date separators).
+ * Convert the data into proper javascript objects
+ *
+ * @param {Array} activityData json formatted activity data
+ * @returns {Array}
+ */
+RoomsJSONAdapter.formatActivity = (activityData) =>
+  activityData.map((activity) => {
+    let formattedActivity = activity;
+
+    if (typeof activity === 'object') {
+      // Clone the object since we will be modifying it
+      formattedActivity = Object.assign({}, activity);
```

Comment:
spread operator is what we have been utilizing for the rest of the implementation.

Reply:
that way a simple ternary operator would do it here, I think?

Reply:
This is to clone the `activity` object so we are not modified the object directly within the `.map` loop.

Reply:
yeah thats what spread operator would do too. 
```js
formattedActivity = {...activity};
```

Reply:
If this is the required way to do it, we need to add it to our eslint rules:
https://github.com/eslint/eslint/pull/9955

Reply:
Sorry, didn't mean for that to come off short. If we have preferred methods of doing something, we should convey that via eslint instead of having to rely on code reviews 👍